models/demos/tg/llama3_70b/demo/demo.py

```diff
@@ -76,6 +76,56 @@ def construct_arg(**kwargs):
     return DemoArgs(model=model_args, tt=tt_args, data=data_args)
 
 
+def nearest_power_of_2(n):
+    return 2 ** (n - 1).bit_length()
+
+
+def get_prompts_for_compilation(tokenized, prompts):
+    # tokenized shape (batch_size, seq_len)
+    tokenized_len = [len(t) for t in tokenized]
+    # pad tokenized_len to be power of 2 and 32 at least
+    padded_tokenized_len = [max(32, nearest_power_of_2(l)) for l in tokenized_len]
+    # Get indexes of first occurences of each unique length
+    unique_lengths = list(set(padded_tokenized_len))
+    # Get indexes of unique_lenghts in tokenized_len
+    indexes = [padded_tokenized_len.index(l) for l in unique_lengths]
+    # Get tokenized and prompts for compilation
+    return [tokenized[i] for i in indexes], [prompts[i] for i in indexes]
+
+
+def demo_warmup(args):
+    # Copy and modify arguments
+    model_args = replace(args.model, num_layers=1)
+    tt_args = replace(args.tt)
+    data_args = replace(args.data, max_output_tokens=2)
+
+    # Skip if model_implementation is not tt
+    if model_args.implementation != "tt":
+        return
+
+    generator = build_generator(model_args, tt_args)
+
+    # Load the model and tokenizer
+    model, tokenizer = generator.model, generator.tokenizer
+
+    tokenized, prompts = load_prompts_file(model_args, data_args, tokenizer)
```

Comment:
Why don't we just use random inputs with the right shapes here? I don't think we need to load the prompts file and get the correctly sizes promps out of there.

Reply:
Since Jack added option for k_chunk_size then if we use it we might not hit precompiled kernel every time. With real shapes we are using same examples. Loading file is super fast and I don't see much of a benefit with using random inputs over data from files.